### PR TITLE
feat: add dynamic mapping schema

### PR DESCRIPTION
**Pull request by natrad100**

### CODE REVIEW

File: src/kappe/utils/msg_def.py [hunk 1]
```diff
@@ -7,30 +7,43 @@
...
 
-    fields = get_message(msg).get_fields_and_field_types()
+    parser_def_class = get_message(msg)
 
     dependencies = []
-    for type_name in fields.values():
-        # primitive
-        if '/' not in type_name:
+    for field_type in get_message_slot_types(parser_def_class).values():
+        # builtin_interfaces are expected to be known by the parser
```
Comment (natrad100) [inline src/kappe/utils/msg_def.py:29]: need this cleaned up, look at the diff.

**Review by natrad100 — COMMENTED** [review]
Just clean that section up and this should be good to go. @KyleJewiss

**Comment by natrad100** [comment]
@KyleJewiss wheres this at

**Comment by KyleJewiss** [comment]
> @KyleJewiss wheres this at

@natrad100 just made the PR on Kappe

**Comment by KyleJewiss** [comment]
I created and merged in this branch. Unlike this branch it does not contain any history of GR related code, and it has deliberately separated commits to make it easier to understand for a merge. 
https://github.com/Greenroom-Robotics/kappe/tree/kylejewiss/msg-interface-upgrader-plugin

A PR has now been made on Kappe with this branch merged into our fork.

**Comment by KyleJewiss** [comment]
The branch was merged into our fork, and that has been submitted to Kappe as a PR here.
https://github.com/sensmore/kappe/pull/5

**Pull request closed without merging**